File: Data Structures & Algorithms/two-integer-sum/submission-3.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        # If the array is sorted use the two pointer method

        # If the array is not sorted you can use the two pointer method: key = number, value = index
        hashMap = {}
        result = []

        for i in range(0,len(nums)):
            diff = target - nums[i]
            if diff in hashMap:
                if hashMap[diff] < i:
                    result.append(hashMap[diff])
                    result.append(i)
                    return result
                else:
                    result.append(i)
                    result.append(hashMap[diff])
                    return result
            
            hashMap[nums[i]] = hashMap.get(nums[i], i)
        
        return result

File: Data Structures & Algorithms/two-integer-sum-ii/submission-3.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int i = 0;
        int j = numbers.length - 1;

        while(i < j && (i < numbers.length) && (j > -1)){
            int calculate = numbers[i] + numbers[j];

            if(calculate == target){
                return new int[] {i+1, j+1};
            }

            else if(calculate < target){
                i++;
            }

            else{
                j--;
            }
        }

        return new int[] {-1, -1};
    }
}
